tests/retriever/test_rig_retriever.py

```diff
@@ -3,101 +3,71 @@
-        return None
-
-    def get_camera_pose(self, index: int) -> Optional[Pose3]:
-        return None
-
-
-class TestConstraint(GtsamTestCase):
-    def setUp(self):
-        """Set up two constraints to test."""
-        aTb = gtsam.Pose3(gtsam.Rot3.Yaw(np.pi / 2), gtsam.Point3(1, 2, 3))
-        cov = gtsam.noiseModel.Isotropic.Variance(6, 1e-3).covariance()
-        counts = np.zeros((5, 5))
-        counts[0][0] = 200
-        self.a_constraint_b = Constraint(1, 2, aTb, cov, counts)
-
-        aTc = gtsam.Pose3(gtsam.Rot3(), gtsam.Point3(0.5, 0.5, 0.5))
-        variances = np.array([1e-4, 1e-5, 1e-6, 1e-1, 1e-2, 1e-3])
-        aCOVc = gtsam.noiseModel.Diagonal.Variances(variances).covariance()
-        counts = np.zeros((5, 5))
-        counts[1][2] = 100
-        self.a_constraint_c = Constraint(1, 3, aTc, aCOVc, counts)
-
-    def test_equals(self):
-        """Test we can test."""
-        self.assertFalse(self.a_constraint_b.equals(self.a_constraint_c, 1e-9))
-
-    def test_read(self):
-        """Read from file in test data directory."""
-        path = DEFAULT_FOLDER / "test_constraints.txt"
-        actual = Constraint.read(str(path))
-        expected = [self.a_constraint_b, self.a_constraint_c]
-        self.gtsamAssertEquals(actual[0], expected[0], 1e-9)
-        self.gtsamAssertEquals(actual[1], expected[1], 1e-9)
-
-    def test_round_trip(self):
-        """Round-trip from temporary file"""
-        with tempfile.TemporaryDirectory() as tempdir:
-            temp_file = Path(tempdir) / "temp_constraints.txt"
-            constraints = [self.a_constraint_b, self.a_constraint_c]
-            Constraint.write(str(temp_file), constraints)
-            actual = Constraint.read(temp_file)
-            self.gtsamAssertEquals(actual[0], constraints[0], 1e-9)
-            self.gtsamAssertEquals(actual[1], constraints[1], 1e-9)
-
-    def test_edges(self):
-        """Test creating edges between cameras."""
-        self.assertEqual(self.a_constraint_b.edges(threshold=30), [(5, 10)])
-        self.assertEqual(self.a_constraint_c.edges(threshold=30), [(5 + 1, 15 + 2)])
+TEST_DATASET_DIR_PATH = DATA_ROOT_PATH / "hilti_exp4_small"
 
 
 class TestRigRetriever(unittest.TestCase):
     def test_rig_retriever(self) -> None:
         """Assert that we can parse a constraints file from the Hilti SLAM team and get constraints."""
 
-        loader = HiltiLoader(DEFAULT_FOLDER)
-        constraints_path = DEFAULT_FOLDER / "test_constraints.txt"
-        retriever = RigRetriever(constraints_path, threshold=30)
+        loader = HiltiLoader(TEST_DATASET_DIR_PATH)
+        retriever = RigRetriever(threshold=30)
 
         pairs = retriever.run(loader=loader)
 
         # We know these to be the right values from setUp() method.
-        self.assertEqual(len(pairs), 2)
-        expected = [(5, 10), (6, 17)]
-        self.assertEqual(pairs[:5], expected)
+        self.assertEqual(len(pairs), 41)
+        expected = [
+            (0, 1),
```

Comment:
I am assuming you crafted this values by hand and hence not checking

Reply:
No, it's real data. I double counted before but I think they are right now.